src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/TimeSpanExtensions.cs

```diff
@@ -72,5 +80,107 @@ public static TimeSpan CalculateBackoffDuration(this TimeSpan minDuration, TimeS
+        {
+            if (attempts < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(attempts), attempts, "The number of attempts should not be less than 1.");
+            }
+
+            if (attempts == 1)
+            {
+                return minDuration;
+            }
+
+            //
+            // IMPORTANT: This can overflow
+            double calculatedMilliseconds = Math.Max(1, minDuration.TotalMilliseconds) * ((long)1 << Math.Min(attempts, MaxAttempts));
+
+            if (calculatedMilliseconds > maxDuration.TotalMilliseconds ||
+                    calculatedMilliseconds <= 0 /*overflow*/)
+            {
+                calculatedMilliseconds = maxDuration.TotalMilliseconds;
+            }
+
+            TimeSpan maxBackoffDuration = CalculateMaxStartupBackoffDuration(startupTimeElapsed);
+
+            TimeSpan calculatedDuration = TimeSpan.FromMilliseconds(calculatedMilliseconds);
+
+            if (calculatedDuration > maxBackoffDuration)
+            {
+                calculatedDuration = maxBackoffDuration;
+            }
+
+            return calculatedDuration.Jitter(StartupJitterRatio);
+        }
+
+        /// <summary>
+        /// This method tries to get the fixed backoff duration for the elapsed startup time if possible.
+        /// </summary>
+        /// <param name="startupTimeElapsed">The time elapsed since the current startup began.</param>
+        /// <param name="backoff">The backoff time span if getting the fixed backoff is successful.</param>
+        /// <returns>A boolean indicating if getting the fixed backoff duration was successful.</returns>
+        public static bool TryGetFixedBackoff(this TimeSpan startupTimeElapsed, out TimeSpan backoff)
+        {
+            if (startupTimeElapsed > FailOverConstants.StartupFixedBackoffDuration)
+            {
+                backoff = TimeSpan.Zero;
+
+                return false;
+            }
+
+            backoff = CalculateMaxStartupBackoffDuration(startupTimeElapsed).Jitter(StartupJitterRatio);
+
+            return true;
+        }
+
+        private static TimeSpan Jitter(this TimeSpan timeSpan, double ratio)
+        {
+            if (ratio < 0 || ratio > 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(ratio));
+            }
+
+            if (ratio == 0)
+            {
+                return timeSpan;
+            }
+
+            var rand = new Random();
+
+            double interval = Math.Abs(timeSpan.TotalMilliseconds * ratio);
+
+            double lowest = timeSpan.TotalMilliseconds - interval * 0.5;
+
+            double offset = interval * rand.NextDouble();
+
+            return TimeSpan.FromMilliseconds(lowest + offset);
+        }
+
+        private static TimeSpan CalculateMaxStartupBackoffDuration(TimeSpan startupTimeElapsed)
```

Comment:
I really only see a need for this logic in TryGetFixedBackoff, and in that case the default return isn't needed since TryGetFixedBackoff would return false. I recommend moving it in there.